Sources/XcodeGenKit/SpecLoader.swift

```diff
@@ -15,9 +15,9 @@ public class SpecLoader {
         self.version = version
     }
 
-    public func loadProject(path: Path) throws -> Project {
+    public func loadProject(path: Path, environmentVariables: [String: String] = ProcessInfo.processInfo.environment) throws -> Project {
```

Comment:
```suggestion
    public func loadProject(path: Path, variables: [String: String] = [:]) throws -> Project {
```
The generate command can pass in `ProcessInfo.processInfo.environment` here https://github.com/yonaskolb/XcodeGen/blob/master/Sources/XcodeGenCLI/GenerateCommand.swift#L61